tsl/src/continuous_aggs/create.c

```diff
@@ -690,10 +693,16 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 	bool found = false;
 
 	/*
-	 * Make sure bucket_width was initialized by caggtimebucketinfo_init().
-	 * This assumption is used below.
+	 * Make sure tbinfo is initialized. This assumption is used below.
+	 *
+	 * It's not guaranteed that caggtimebucketinfo_init() will be called before
```

Comment:
If it's not guaranteed, doesn't this mean that `tbinfo->origin` may be overwritten by `caggtimebucketinfo_init()` later?

Reply:
To be honest I'm a bit confused by the way we call caggtimebucketinfo_init(). Here is how cagg_validate_query() is implemented:

```
...
if (rte->relkind == RELKIND_RELATION) {
   ...
   caggtimebucketinfo_init(...)
   ...
}

...

caggtimebucket_validate(...)
```

So currently we can call validate() without init() but not wise versa. Also (1) the name "_init()" seems to be wrong, it just fills certain fields (2) I used Asserts() before I discovered the fact stated above and it seems that IN PRACTICE init() is always called before validate().

So, do you suggest we use Asserts() here or keep the assignment? Refactoring cagg_validate_query() is another option but I think we better do this in a separate pull request.

Reply:
OK, after thinking a bit more about it I returned Assert()'s here as it is currently in `master`. I think the code is more clear this way. Please feel free to re-open the discussion if necessary.